contracts/liquidator/LiquidatorV2.sol

```diff
@@ -0,0 +1,235 @@
+    address public immutable recipient;
+
+    /// @notice The scale for asset price calculations
+    uint256 public constant QUOTE_PRICE_SCALE = 1e18;
+
+    constructor(
+        CometInterface _comet,
+        address _factory,
+        address _WETH9,
+        address _recipient
+    ) PeripheryImmutableState(_factory, _WETH9) {
+        comet = _comet;
+        recipient = _recipient;
+    }
+
+    function balanceOfAsset(address asset) internal returns (uint256, uint256) {
+        uint256 collateralBalance = comet.getCollateralReserves(asset);
+
+        uint256 baseScale = comet.baseScale();
+
+        uint256 quotePrice = comet.quoteCollateral(asset, QUOTE_PRICE_SCALE * baseScale);
+        uint256 collateralBalanceInBase = baseScale * QUOTE_PRICE_SCALE * collateralBalance / quotePrice;
+
+        return (
+            collateralBalance,
+            collateralBalanceInBase
+        );
+    }
+
+    /**
+     * @notice XXX
+     * @dev for use with static call
+     */
+    function availableCollateral(
+        address[] calldata liquidatableAccounts
+    ) external returns (address[] memory, uint256[] memory, uint256[] memory) {
+        comet.absorb(address(this), liquidatableAccounts);
+
+        uint8 numAssets = comet.numAssets();
+
+        address[] memory assets = new address[](numAssets);
+        uint256[] memory collateralReserves = new uint256[](numAssets);
+        uint256[] memory collateralReservesInBase = new uint256[](numAssets);
+
+        for (uint8 i = 0; i < numAssets; i++) {
+            address asset = comet.getAssetInfo(i).asset;
+            assets[i] = asset;
+            uint256 collateralBalance = comet.getCollateralReserves(asset);
+
+            collateralReserves[i] = collateralBalance;
+
+            uint256 baseScale = comet.baseScale();
+
+            // XXX check that reserves are still under target reserves?
+            uint256 quotePrice = comet.quoteCollateral(asset, QUOTE_PRICE_SCALE * baseScale);
+            collateralReservesInBase[i] = baseScale * QUOTE_PRICE_SCALE * collateralBalance / quotePrice;
+        }
+
+        return (
+            assets,
+            collateralReserves,
+            collateralReservesInBase
+        );
+    }
+
+    // XXX move
+    struct FlashCallbackData {
+        uint256 flashLoanAmount;
+        PoolAddress.PoolKey poolKey;
+        address[] assets;
+        uint256[] assetBaseAmounts;
+
+        address[] swapTargets;
+        bytes[] swapTransactions;
+    }
+
+    // XXX name?
+    function absorbAndArbitrage(
+        address[] calldata liquidatableAccounts,
+        address[] calldata assets,
+        // uint256[] calldata assetBaseAmounts,
+        address[] calldata swapTargets,
+        bytes[] calldata swapTransactions
+    ) external {
+        comet.absorb(address(this), liquidatableAccounts);
+        emit Absorb(msg.sender, liquidatableAccounts);
+
+        address poolToken0 = 0x6B175474E89094C44Da98b954EedeAC495271d0F; // XXX DAI
```

Comment:
- [ ] remove hard-coded DAI reference
- [ ] build pool on contract construction
- [ ] potentially allow the pool to be re-set